smarty/sampler.py

```diff
@@ -97,6 +97,21 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
         # Save bond list to use throughout
         bondset = [("-","singly"), ("=", "doubly"), ("#","triply"), (":", "aromatic")]
 
+        used_basetypes = list()
+        self.atomtypes_with_no_matches = set()
+        # Check all SMART strings that are used as a base type
+        for (smarts, atom_type) in self.basetypes:
+            check_flag = self.smarts_matches(smarts)
+            if check_flag:
+                # Keep used base types
+                used_basetypes.append( ( smarts, atom_type) )
+            else:
+                # Remove unused base types
+                self.atomtypes_with_no_matches.add( smarts )
+        self.basetypes = copy.deepcopy(used_basetypes)
+        if verbose:
+            print("USED BASE TYPES: %s" % self.basetypes)
```

Comment:
@camizanette We should show what base types are kept somewhere though. probably something like

```
if verbose:
    print("USED BASE TYPES:")
    for (smarts, typename) in self.basetypes:
        print("%10s %25s" % (smarts, typename))
```